realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsMongoCollection.cpp

```diff
@@ -116,41 +116,36 @@ Java_io_realm_internal_objectstore_OsMongoCollection_nativeCount(JNIEnv* env,
-        collection->find_one(filter, JavaNetworkTransport::create_result_callback(env, j_callback, collection_mapper_find_one));
-    }
-    CATCH_STD()
-}
-
-JNIEXPORT void JNICALL
-Java_io_realm_internal_objectstore_OsMongoCollection_nativeFindOneWithOptions(JNIEnv* env,
-                                                                              jclass,
-                                                                              jlong j_collection_ptr,
-                                                                              jstring j_filter,
-                                                                              jstring j_projection,
-                                                                              jstring j_sort,
-                                                                              jlong j_limit,
-                                                                              jobject j_callback) {
-    try {
-        auto collection = reinterpret_cast<RemoteMongoCollection*>(j_collection_ptr);
-        uint64_t limit = std::uint64_t(j_limit);
 
-        bson::BsonDocument filter(JniBsonProtocol::parse_checked(env, j_filter, Bson::Type::Document, "BSON filter must be a Document"));
-        bson::BsonDocument projection(JniBsonProtocol::parse_checked(env, j_projection, Bson::Type::Document, "BSON projection must be a Document"));
-        bson::BsonDocument sort(JniBsonProtocol::parse_checked(env, j_sort, Bson::Type::Document, "BSON sort must be a Document"));
-        RemoteMongoCollection::RemoteFindOptions options = {
-                limit,
-                projection,
-                sort
-        };
-
-        collection->find_one(filter, options, JavaNetworkTransport::create_result_callback(env, j_callback, collection_mapper_find_one));
+        switch (j_find_one_type) {
+            case io_realm_internal_objectstore_OsMongoCollection_FIND_ONE:
+                collection->find_one(filter, JavaNetworkTransport::create_result_callback(env, j_callback, collection_mapper_find_one));
+                break;
+            case io_realm_internal_objectstore_OsMongoCollection_FIND_ONE_WITH_OPTIONS:
+                uint64_t limit = std::uint64_t(j_limit);
+
+                bson::BsonDocument projection(JniBsonProtocol::parse_checked(env, j_projection, Bson::Type::Document, "BSON projection must be a Document"));
+                bson::BsonDocument sort(JniBsonProtocol::parse_checked(env, j_sort, Bson::Type::Document, "BSON sort must be a Document"));
+                RemoteMongoCollection::RemoteFindOptions options = {
+                        limit,
+                        projection,
+                        sort
+                };
+
+                collection->find_one(filter, options, JavaNetworkTransport::create_result_callback(env, j_callback, collection_mapper_find_one));
+                break;
+        }
```

Comment:
Maybe throw an error if it is an unexpected type, as you do in some of the other similar methods.

Reply:
Yes, totally forgot about it.